### DISCUSSION TITLE
[v5] Email verification with auth token

### DISCUSSION THREAD

**Discussion by rochagbr** [adonisjs/core#2297] [Help]
Is there a good way to do this?
For password reset, for example.

I am currently using the token name to do this

Is [Adonis Persona](https://github.com/adonisjs/persona) a way to consider on V5?

**Comment by aayush123** [comment]
You should probably elaborate your use case a little more

**Comment by anthrogan** [comment]
The logic behind this sort of stuff isn't really adonis or any other framework, its more a process you implement. 

The basic way to do password reset with JWT is, when a user requests a password reset a post is made to the user table with a "reset token" and stored with the users account identified by email. This is also attached to the reset URL in email (https://exampletest.com/reset/password?token=resetToken) which is sent to the users personal email address. Once a reset is complete the token in the email link is used to identify what user password to change.

There are lots of guides on the internet that can take you through this process. Similar happens with verification of accounts, just swop reset for verification
